data-loader/data/images.js

```diff
@@ -0,0 +1,113 @@
+import sharp from "sharp";
+import { Binary } from 'mongodb';
+const binaryCachedImages = {};
+
+export async function init() {
+    const cachedImages = {
+        image: [],
+        large: [],
+        medium: [],
+        small: [],
+        thumbnail: []
+    }
+    console.log("Caching conversions");
+    for (let i = 0; i < 20; i++) {
+        const r = Math.floor(Math.random() * 256);
+        const g = Math.floor(Math.random() * 256);
+        const b = Math.floor(Math.random() * 256);
+        cachedImages.image.push(await sharp({
+            create: {
+                width: 600,
+                height: 600,
+                channels: 3,
+                background: { r, g, b }
+            }
+            })
+            .jpeg()
+            .toBuffer());
+        cachedImages.small.push(await sharp({
+            create: {
+                width: 235,
+                height: 235,
+                channels: 3,
+                background: { r, g, b }
+            }
+            })
+            .png()
+            .toBuffer());
+        cachedImages.thumbnail.push(await sharp({
+            create: {
+                width: 100,
+                height: 100,
+                channels: 3,
+                background: { r, g, b }
+            }
+            })
+            .png()
+            .toBuffer());
+    }
+    cachedImages.large = cachedImages.image;
+    cachedImages.medium = cachedImages.image;
+    Object.keys(cachedImages).forEach((key) => {
+        binaryCachedImages[key] = cachedImages[key].map(Binary);
+    });
+}
+
+
+export function getImage(storeName, imageIndex) {
+    return binaryCachedImages[storeName][imageIndex];
+}
+
+/**
+ * @method generateImage
```

Comment:
Can we port this function into the main, plugin version of devtools? The colored images are way better than the static ones